### ISSUE TITLE
How should we do versioning?

### ISSUE THREAD

**Issue by magnuhar** [wemadefree/WeBook#35]
One of the questions we haven't discussed yet is versioning, and how we want to do this. 
Maybe this could be a issue for project leader?

Design wise i would like to have the version number reflected on the footer of the sidenav. Perhaps we should add an environment variable, "SHOW_VERSION", to toggle this on or off at will. Maybe also at login?
On a related note, it would also be prudent to have a flag indicating if an instance of the application is live or production. Developer accounts should thusly be warned if logged in on a production site -- could probably be segregated only to the admin panel.

**Comment by eivl** [comment]
Lets discuss and solve this. 
`APP_VERSION` is added to a branch for testing, i think we could just do that everywhere and use the .env file for versioning. 
We already have a flag for it the django debug flag. and for the segregation that is another issue. 

We can choose between Semantic versioning: that follows this rule.
`MAJOR.MINOR.PATCH` and sometimes you would add a `QUALIFIER` at the end. 

![EXAMPLE](https://miro.medium.com/max/1036/1*Qx9SMeVZoaSoVCVLAHiIYw.png)

This is great for developers, and not great for users. 

The other i dont remember the real name, i call it `Calendar Versioning` where you have this format. 
`YEAR.RELEASE.BUILD`  and optional `STATUS`

![EXAMPLE](https://miro.medium.com/max/1400/0*djbnhOT72Kp0NG8D.png)

This is great for users because they get a feel to what the version means, it is also enough for developers since you can make it more verbose by adding context to build and status.


I suggest we go for calendar versioning for this reason.

**Comment by magnuhar** [comment]
I agree with calendar versioning. Is there a way to automate this perhaps? So that when one merges to (when we're in the production setup) main it would "tag" a new version? Or is this something we need to handle manually on each merge to main/prod branch?

**Comment by eivl** [comment]
Yes, it is a way, I have seen it done, I just don't know-how. I will find the solution to how we can do it.
